domainconnectzone/DomainConnect.py

```diff
@@ -130,13 +131,14 @@ def resolve_variables(input_, domain, host, params, recordKey):
             value = host
         elif name in params:
             value = params[name]
-        else:
+
+        if value is None:
             raise MissingParameter("No value for parameter '" + name + "'")
 
         # Place the value into the input string
         input_ = input_.replace('%' + name + '%', value)
```

Comment:
This line will raise a TypeError if `null` is passed in as a JSON value. This is currently detected by domain-attach-api as an HTTP 500 (rather than MissingParameter which will be detected as an HTTP 400).

```
{
    "force": false,
    "dryrun": true,
    "providerId": "exampleservice.domainconnect.org",
    "serviceId": "template1",
    "params": {
    	"IP": "1.2.3.4",
    	"RANDOMTEXT": null
    }
}
```

```
TypeError: "coercing to Unicode: need string or buffer, NoneType found"
```